gateway/models/users.go

```diff
@@ -0,0 +1,127 @@
+package models
+
+import (
+	"errors"
+
+	"github.com/hoophq/hoop/common/log"
+	"gorm.io/gorm"
+)
+
+type User struct {
+	ID             string
+	OrgID          string
+	Subject        string
+	Name           string
+	Picture        string
+	Email          string
+	Verified       bool
+	Status         string
+	SlackID        string
+	HashedPassword string
+}
+
+func ListUsers(orgID string) ([]User, error) {
+	log.Debugf("listing users for org=%s", orgID)
+	var users []User
+	if err := DB.Where("org_id = ?", orgID).Find(&users).Error; err != nil {
+		log.Errorf("failed to list users, reason=%v", err)
+		return nil, err
+	}
+
+	return users, nil
+}
+
+func GetInvitedUserByEmail(email string) (*User, error) {
+	log.Debugf("getting invited user=%s", email)
+	var user *User
+	if err := DB.Where("email = ? AND status = ?", email, "invited").First(&user).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return user, nil
+}
+
+func GetUserByOrgIDAndSlackID(orgID, slackID string) (*User, error) {
+	log.Debugf("getting user=%s for org=%s", slackID, orgID)
+	var user *User
+	if err := DB.Where("org_id = ? AND slack_id = ?", orgID, slackID).First(&user).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return user, nil
+}
+
+func GetUserBySubjectAndOrg(subject, orgID string) (*User, error) {
+	log.Debugf("getting user=%s for org=%s", subject, orgID)
+	var user *User
+	if err := DB.Where("org_id = ? AND subject = ?", orgID, subject).First(&user).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return user, nil
+}
+
+func GetUserByEmailAndOrg(email, orgID string) (*User, error) {
```

Comment:
Let's try to keep a pattern where the orgID comes first since most of the resources derive from an organization. Therefore, there's no need to call the function of `...EmailAndOrg`, it's implicit if you're passing this information:

```go
func GetUserByEmail(orgID, email string) (*User, error) {
  (...)
}
```